From db0640c90104b48f4e9f67d023f65d87fdb86a13 Mon Sep 17 00:00:00 2001
From: Guo Ren <guoren@linux.alibaba.com>
Date: Tue, 4 May 2021 20:51:11 +0800
Subject: [PATCH 66/72] riscv: thead: vector: Fixup compile error with new
 assembler

New assembler couldn't support rd = rs, so change the asm code for
compatible.

Signed-off-by: Guo Ren <guoren@linux.alibaba.com>
---
 arch/riscv/kernel/vector.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/riscv/kernel/vector.S b/arch/riscv/kernel/vector.S
index 0afc308a6e02..c7f45e1821cd 100644
--- a/arch/riscv/kernel/vector.S
+++ b/arch/riscv/kernel/vector.S
@@ -77,7 +77,7 @@ ENTRY(__vstate_restore)
 	csrw	CSR_VXRM, t0
 	ld	t0,  TASK_THREAD_VL_V0(a0)
 	ld	t2,  TASK_THREAD_VTYPE_V0(a0)
-	vsetvl	t0, t0, t2
+	vsetvl	t3, t0, t2
 
 	csrc	sstatus, t1
 	ret
-- 
2.17.1

